File: Level2/Materal.Workflow/Src/Materal.Workflow.Vue/src/scripts/StepDatas/BranchStepData.ts
import { IStepData } from "./Base/IStepData";
import { StepData } from "./Base/StepData";

export class BranchStepData extends StepData {
    /**
     * 计划节点
     */
    public StepDatas: IStepData[] = [];
    /**
     * 下一步
     */
    public Next?: IStepData;
    constructor(id: string) {
        super(`${BranchStepData.name}`, "分支节点", id);
    }
}

File: Level3/Materal.Workflow/Src/Materal.Workflow/StepDatas/BranchStep/BranchStepData.cs
using Newtonsoft.Json;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Materal.Workflow.StepDatas
{
    /// <summary>
    /// 并行节点数据
    /// </summary>
    public class BranchStepData : StepData, IStepData
    {
        /// <summary>
        /// 节点数据组
        /// </summary>
        [Required, MinLength(1)]
        public List<IStepData> StepDatas { get; set; } = new();
        /// <summary>
        /// 下一步
        /// </summary>
        [System.Text.Json.Serialization.JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
        public IStepData? Next { get; set; }
    }
}
